desugarer.go

```diff
@@ -253,7 +270,7 @@ func buildLiteralString(value string) ast.Node {
 func buildSimpleIndex(obj ast.Node, member ast.Identifier) ast.Node {
 	return &ast.Index{
 		Target: obj,
-		Id:     &member,
+		Index:  buildLiteralString(string(member)),
```

Comment:
ast.Identifier is immutable though isn't it?

Reply:
The underlying data of strings (and `ast.Identifier` which is the same thing underneath) is immutable. But it's possible to do something like `a := "foo"; a = "bar"`. 
 
The `string(member)` is there because `string` and `ast.Identifier` need to be explicitly converted, even though they are really the same thing underneath.

(And the point of changing this line is to build desugared node in one step, rather than something that needs to be desugared).